### DISCUSSION TITLE
AuthData length 39

### DISCUSSION THREAD

**Discussion by krzysztof90** [Yubico/libfido2#828] [Q&A] [answered]
I am using Fido2Net library.
Why AuthData is returning 39 bytes instead of 37?
I can't proceed with my verification because of that. When I skip 2 first bytes it verifies rp correctly but signature verification fails.
It calls fido_assert_authdata_ptr and fido_assert_authdata_len methods

**Comment by LDVG** [comment, marked as answer]
`fido_assert_authdata_ptr()` returns the CBOR-encoded authenticator data, i.e. a CBOR byte string. For 37 bytes of payload, this will be an extra tag plus length byte. If you instead need the "raw" authenticator data, use `fido_assert_authdata_raw_{ptr,len}`.
